src/services/item/plugins/published/constants.ts

```diff
@@ -3,8 +3,12 @@ import { Context } from '@graasp/sdk';
 import { CLIENT_HOSTS } from '../../../../utils/config';
 import { Item } from '../../entities/Item';
 
-export const buildPublishedItemLink = (item: Item) => {
-  const library = CLIENT_HOSTS.find(({ name }) => name === Context.Library);
-  const target = new URL(item.id, library?.url);
+export const buildPublishedItemLink = (item: Item): string | null => {
+  const library = CLIENT_HOSTS.find(({ name }) => name === Context.Library)?.url;
+  if (!library) {
+    return null;
+  }
+
+  const target = new URL(`/${item.id}`, library.origin);
```

Comment:
You don't need to use the `origin` property since you can use the URL type directly.